mailgun-relayery/src/servers/control/pages.go

```diff
@@ -0,0 +1,81 @@
+package control
+
+import (
+	"math"
+	"servers/protoed_channel"
+	"sort"
+)
+
+// ChannelPageResponse computes the response for a valid channels page listing request.
+//
+// ChannelPageResponse requires:
+// * page > 0
+// * perPage > 0
+// * perPage * (page - 1) <= int32(len(channelsMap))
+//
+// ChannelPageResponse ensures:
+// * channelsPage.Page == page
+// * channelsPage.PerPage == perPage
+// * channelsPage.PageCount >= 0
+// * !(len(channelsMap) > 0) || len(channelsPage.Channels) > 0
```

Comment:
I wrote the condition like this instead of ´len(channelsMap) == 0 || len(channelsPage.Channels) > 0´ to better convey the idea of the logical implication (and the same one line below)